content/docs/for-buildpack-authors/how-to/write-buildpacks/specify-env.md

```diff
@@ -3,15 +3,160 @@ title="Specify the environment"
 weight=99
 +++
 
+Environment variables are a common way to configure various buildpacks at build-time, make executables available on the `$PATH` environment variable, and configure the app at runtime.
+
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+When `clear-env` is not set to `true`, the `lifecycle` MUST set user-provided environment variables in the environment of `/bin/detect` or `/bin/build` such that:
+
+* For layer path environment variables, user-provided values are prepended before any existing values and are delimited by the OS path list separator.
+* For all other environment variables, user-provided values override any existing values.
+* The environment variable prefix `CNB_` is reserved. It MUST NOT be used for environment variables that are not defined in this specification or approved extensions.
+
+### POSIX Path Variables
+
+When the `lifecycle` runs buildpacks, it first tears down anything defined on the `build-time` base image of the environment. It only allows a [specific set](https://github.com/buildpacks/lifecycle/blob/a43d5993a4f2cc23c44b6480ba2ab09fe81d57ed/env/build.go#L9-L19) of pre-configured environment variables through.
+
+Then, it applies `buildpack defined` environment variables including anything that might have been configured by a buildpack that ran earlier during the `build` phase.
+
+>Note that buildpacks cannot set environment variables for other buildpacks during the `detect` phase.
+
+#### Example
+
+Looking at the following directory tree and table
+
+```text
+
+layers/
+└── some-buildpack-id
+    ├── some-build-layer
+    │   ├── bin
+    │   │   └── some-binary
+    │   ├── env
+    │   │   └── SOME_VAR # contents foo
+    │   ├── env.build
+    │   │   └── SOME_VAR # contents bar
+    │   └── lib
+    │       ├── some-shared-library
+    │       └── some-static-library
+    └── some-build-layer.toml
+
```

Comment:
Let's add some explanatory text here

Reply:
With this tree:
* The current buildpack will see `SOME_VAR=foo` in its environment
* The current buildpack will find `some-binary` in `PATH`
* The current buildpack will find `some-static-library` in `LIBRARY_PATH`

Thus, any `<layers>/<layer>/<env>` directory is for setting environment variables directly, and `<layers>/<layer>/<bin>`, `<layers>/<layer>/<lib>`, etc. offer a convenient way to modify POSIX path variables. The full list of convenience directories is summarized in the table below:

Reply:
added